system/alternatives.py

```diff
@@ -114,7 +124,16 @@ def main():
                         link = line.split()[1]
                         break
 
-    if current_path != path:
+    if not force:
+        need_to_force = False
+    elif not os.path.islink(link):
```

Comment:
I haven't done any testing against this, but I think I'm with @mulby here. Rather than try to be intelligent about whether or not force should be used, we should trust the user's suggestion.

@jamespharaoh are there any problems with that approach?